### PR TITLE
✨ Feature: Dependency-diff ecosystem naming convention mapping (GitHub -> OSV)

### LINKED ISSUE #2078: Feature: map the current Dependencydiff package naming convention (GitHub) to OSV
**Is your feature request related to a problem? Please describe.**
In v0 of the API/CLI Dependencydiff (PR #2046/PR #2077), the formats (convention) of dependency fields like the package `ecosystem` (GH uses `pypi` for python pkgs whereas OSV uses `pip`) follow the GitHub naming convention. Examples: 
```json
[
  {
    "changeType": "added",
    "packageUrl": "pkg:golang/github.com/google/go-github/v35@35.2.0",
    "sourceRepository": null,
    "manifestPath": "go.mod",
    "ecosystem": "gomod",
    "version": "35.2.0",
    "name": "github.com/google/go-github/v35",
  },
  {
    "changeType": "updated",
    "packageUrl": "pkg:golang/gocloud.dev@0.23.0",
    "sourceRepository": "https://github.com/google/go-cloud",
    "manifestPath": "go.mod",
    "ecosystem": "gomod",
    "version": "0.23.0",
    "name": "gocloud.dev",
  },
]
```

**Describe the solution you'd like**
By @laurentsimon: since we may use a different Dependency-diff source API in the future, it would be nice not to get stuck using GH naming convention. 

I can do the `ecosystem` mapping using[ this list](https://ossf.github.io/osv-schema/#affectedpackage-field) (near **The defined ecosystems are:**), by defining a static map in the scorecard code to convert the GH ecosystems to the OSV ones. I'll use a follow-up PR to do this mapping very soon.

### PR DESCRIPTION
**Pull request by aidenwang9867**
#### What kind of change does this PR introduce?
1. Dependency ecosystem naming convention: map ecosystem names using the[ GitHub naming convention](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems) into names using the [OSV naming convention](https://ossf.github.io/osv-schema/#affectedpackage-field).

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
`GetDependencydiffResults` returns depdency-diff check results with package ecosystem names in the GitHub naming convention.

#### What is the new behavior (if this is a feature change)?**
GH -> OSV.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
Fixes #2078

#### Special notes for your reviewer
**Please LGTM and merge this PR first,** then #2077. This one optimizes the Dependency-diff API and **has higher priority**.

#### Does this PR introduce a user-facing change?
No.

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

File: dependencydiff/mapping.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	"fmt"
+)
+
+// EcosystemOSV is a package ecosystem supported by OSV.
+type ecosystemOSV string
+
+// Data source: https://ossf.github.io/osv-schema/#affectedpackage-field
+const (
+	// The Go ecosystem.
+	goOSV ecosystemOSV = "Go"
```
Comment (laurentsimon) [inline dependencydiff/mapping.go:27]: let's just call it `ecosystem`. The comment is enough without the need to bloat the variables names.
You can name each variable `ecosystemGo`, `ecosystemNpm`, etc

File: dependencydiff/mapping.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	hexOSV = "Hex" // nolint:unused
+)
+
+// ecosystemGitHub is a package ecosystem supported by GitHub.
+type ecosystemGitHub string
+
+// Data source: https://docs.github.com/en/code-security/supply-chain-security/understanding-
+// your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems
+// nolint
+const (
```
Comment (laurentsimon) [inline dependencydiff/mapping.go:73]: Can we simply this to just a hash map? We just care about a mapping between our ecosystem and a string from GH, and the hashmap is clear enough for the reader.

**Review by laurentsimon — COMMENTED** [review]
Thanks.
The PR does a lot more than the title suggest. Try do do one thing as per tiel in each PR to simplify review.

**Comment by aidenwang9867** [comment]
> Thanks. The PR does a lot more than the title suggest. Try do do one thing as per tiel in each PR to simplify review.

I'll separate it into two PRs: (1) the mapping and (2) the optimizations (var name, logic) in my next push, thanks!

File: dependencydiff/mapping.go (hunk 1 above)
Comment (aidenwang9867) [reply dependencydiff/mapping.go:27]: done & resolved, thanks!

File: dependencydiff/mapping.go (hunk 2 above)
Comment (aidenwang9867) [reply dependencydiff/mapping.go:73]: done & resolved, thanks!

**Review by laurentsimon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**